File: MCP服务器/finance/QuantConnect_mcp-server_3a656b0_python/tests/algorithms/parameter_optimization.py
# region imports
from AlgorithmImports import *
# endregion


class ParameterOptimizationTestAlgorithm(QCAlgorithm):

    def initialize(self):
        self.set_start_date(2010, 1, 1)
        self.set_end_date(2025, 1, 1)
        self._equity = self.add_equity("SPY", Resolution.DAILY)
        self.settings.automatic_indicator_warm_up = True
        self._sma_slow = self.sma(
            self._equity.symbol, self.get_parameter('sma_slow', 21)
        )
        self._sma_fast = self.sma(
            self._equity.symbol, self.get_parameter('sma_fast', 5)
        )

    def on_data(self, data: Slice):
        if (not self._equity.holdings.is_long and 
            self._sma_fast > self._sma_slow):
            self.set_holdings(self._equity.symbol, 1)
            return
        if (not self._equity.holdings.is_short and 
            self._sma_fast < self._sma_slow):
            self.set_holdings(self._equity.symbol, -1)

File: MCP服务器/finance/QuantConnect_mcp-server_3a656b0_python/tests/algorithms/ParameterOptimization.cs
#region imports
    using QuantConnect;
    using QuantConnect.Algorithm;
    using QuantConnect.Indicators;
    using QuantConnect.Data;
    using QuantConnect.Securities.Equity;
#endregion
public class ParameterOptimizationTestAlgorithm : QCAlgorithm
{
    private Equity _equity;
    private SimpleMovingAverage _smaSlow;
    private SimpleMovingAverage _smaFast;

    public override void Initialize()
    {
        SetStartDate(2010, 1, 1);
        SetEndDate(2025, 1, 1);
        
        _equity = AddEquity("SPY", Resolution.Daily);
        Settings.AutomaticIndicatorWarmUp = true;
        _smaSlow = SMA(_equity.Symbol, GetParameter("sma_slow", 21));
        _smaFast = SMA(_equity.Symbol, GetParameter("sma_fast", 5)); 
    }

    public override void OnData(Slice data)
    {
        if (!_equity.Holdings.IsLong && _smaFast > _smaSlow)
        {
            SetHoldings(_equity.Symbol, 1);
            return;
        }
        if (!_equity.Holdings.IsShort && _smaFast < _smaSlow)
        {
            SetHoldings(_equity.Symbol, -1);
        }
    }
}
